src/components/cc-logs-instances/cc-logs-instances.js

```diff
@@ -0,0 +1,653 @@
+        </div>
+      </div>
+    `;
+  }
+
+  /**
+   * @param {Array<Instance>} instances
+   */
+  _renderDeletedInstances (instances) {
+    if (instances.length === 0) {
+      return '';
+    }
+
+    return html`
+      <div class="section section--deleted" role="group" aria-labelledby="deleted-instances">
+        ${this._renderHeader({
+          id: 'deleted-instances',
+          title: i18n('cc-logs-instances.deleted.header'),
+          icon: iconHeaderDeleted,
+        })}
+        ${this._renderInstancesGroupedByDeployment(instances, true)}
+      </div>
+    `;
+  }
+
+  _renderHeader ({ id, title, icon, commit }) {
+    return html`
+      <div class="section-header" id="${ifDefined(id ?? undefined)}">
+        ${icon != null ? html`
+          <cc-icon class="section-header-icon" .icon=${icon}></cc-icon>
+        ` : ''}
+        <span class="section-header-title">${title}</span>
+        ${commit != null ? this._renderCommit(commit) : ''}
+      </div>
+    `;
+  }
+
+  /**
+   * @param {Array<Instance>} instances
+   * @param {boolean} renderState
+   */
+  _renderInstancesGroupedByDeployment (instances, renderState) {
+    const groups = groupBy(instances, (instance) => instance.deployment?.id);
+
+    return html`
+      <div class="section-content deployments">
+        ${Object.entries(groups)
+          .sort(([, instances1], [, instances2]) => instances2[0].deployment.creationDate - instances1[0].deployment.creationDate)
+          .map(([_, instances]) => {
+            return html`
+            <div class="deployment" role="group" aria-labelledby="${instances[0].deployment.id}">
+              ${this._renderDeploymentDetails(instances[0].deployment)}
+              <div class="instances">
+                ${this._renderInstances(instances, renderState)}
+              </div>
+            </div>
+          `;
+          })}
+      </div>
+    `;
+  }
+
+  /**
+   * @param {Array<Instance>} instances
+   * @param {boolean} renderState
+   */
+  _renderInstances (instances, renderState) {
+    return html`
+      ${instances.sort(INSTANCE_SORT_ORDER).map((instance) => this._renderInstance(instance, renderState))}
+    `;
+  }
+
+  /**
+   *
+   * @param {Instance} instance
+   * @param {boolean} renderState
+   */
+  _renderInstance (instance, renderState) {
+    return html`
+      <label class="instance">
+        <input type="checkbox"
+               id="${instance.id}"
+               .checked=${this._isSelected(instance.id)}
+               @change=${this._onInstanceClick}>
+          ${renderState ? this._renderInstanceState(instance) : ''}
+          <span class="instance-index">[${instance.index}]</span>
+          <span class="instance-name">${instance.name}</span>
+          ${instance.kind === 'BUILD' ? html`
+            <cc-icon .icon=${iconBuildInstance}
+                     a11y-name="${i18n('cc-logs-instances.build.a11y-name')}"
+                     class="instance state"></cc-icon>
```

Comment:
**🔨 issue:** `<cc-icon>` has an `instance` CSS class, as well as the wrapping `<label>`. There's no effect right now but I think it's not intended and issues could arise later. I think it's safe to simply remove it.

```suggestion
            <cc-icon .icon=${iconBuildInstance}
                     a11y-name="${i18n('cc-logs-instances.build.a11y-name')}"
                     class="state"></cc-icon>
```